### PR TITLE
Feature#3 responsive menu design

### LINKED ISSUE #3: Responsive menu design
Program responsive menu  (mobile-first approach)

Depends on #1 

### Design
Desktop
Captura de Pantalla 2021-08-26 a la(s) 10 31 11

Tablet
Captura de Pantalla 2021-08-26 a la(s) 10 30 37

Mobile
Captura de Pantalla 2021-08-26 a la(s) 10 30 12

imagen

### Criteria acceptance

[ ] Responsive behavior (desktop, tablet, and mobile)

### PR DESCRIPTION
**Pull request by SampabloC**
# Description
What did you implemented/Why did you implemented this:
 - Add navbar responsive
 - Mobile, Tablet and Desktop version
 - Change logos when user open and close navbar on mobile and tablet version

Resolve #3 

  ## Screenshots 
Mobile Version
imagen
imagen

Tablet Version
imagen
imagen

Desktop Version
imagen

### CODE REVIEW

File: app/views/layouts/_navbar.html.erb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+        if( imageIndex == 0) {
+          logoId.className = 'logo-open';
+        } else {
+          logoId.className = 'logo-closed';
+        }
+        imageIndex = (imageIndex + images.length + 1) % (images.length);
+        $("#logo").attr('src', logos[imageIndex]);
+        $("#image").attr('src', images[imageIndex]);
+      });
+    });
```
Comment (YilKanda) [inline app/views/layouts/_navbar.html.erb:29]: In this part I have two comments:

1. It is preferable not to use JQuery, you can implement this part with turbolinks:

```suggestion
  document.addEventListener("turbolinks:load", function() {
    let images = ["/assets/list.svg", "/assets/close.svg"];
    let logos = ["assets/BCnameLogo.png", "assets/BCsingleLogo.png"];
    let imageIndex = 0;
    let logoId = document.getElementById("logo");
    document.getElementByClassName("navbar-toggler").on("click", function() {
      if (imageIndex == 0) {
        logoId.className = 'logo-open';
      } else {
        logoId.className = 'logo-closed';
      }
      imageIndex = (imageIndex + images.length + 1) % (images.length);
      document.getElementById("logo").attr('src', logos[imageIndex]);
      document.getElementById("image").attr('src', images[imageIndex]);
    });
  });
```

2. This section should be able to be called as a partial from a separate file, so as not to overload this file.

File: app/views/layouts/_navbar.html.erb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+  </script>
+</head>
+<body>
+  <nav class="navbar navbar-expand-lg navbar-light ">
+    <div class="container-fluid">
+      <div class="logo">
+        <img src="/assets/BCnameLogo.png" alt="BCLogo" id="logo" class="logo-closed">
```
Comment (YilKanda) [inline app/views/layouts/_navbar.html.erb:36]: I think this part can take advantage of the use of the Rails image_tag.

File: app/views/layouts/_navbar.html.erb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      </div>
+      <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarTogglerDemo02"
+        aria-controls="navbarTogglerDemo02" aria-expanded="false" aria-label="Toggle navigation" style="border: none; box-shadow: none">
+        <img src="/assets/list.svg" id="image">
```
Comment (YilKanda) [inline app/views/layouts/_navbar.html.erb:40]: I think this part can take advantage of the use of the Rails image_tag.

File: app/views/layouts/_navbar.html.erb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+          <li class="nav-item">
+            <a class="nav-link" aria-current="page" href="#" id="link">Convocatorias</a>
+          </li>
+          <li class="nav-item" id="borderContacto">
+            <a class="nav-link" aria-current="page" href="#" id="linkContacto">Contacto</a>
+          </li>
+          
+        </div>
+          <div class="group609">
+            <img src="/assets/Group 609.svg" alt="group609">
```
Comment (YilKanda) [inline app/views/layouts/_navbar.html.erb:62]: I think this part can take advantage of the use of the Rails image_tag.

**Review by YilKanda — CHANGES_REQUESTED** [review]
The JS code placed in the `_navbar.html.erb` file, I think it is more convenient to put it in a separate file.

File: app/views/layouts/_navbar.html.erb (hunk 1 above)
Comment (xFerchoVJ) [reply app/views/layouts/_navbar.html.erb:29]: I agree with Yil.

**Review by xFerchoVJ — DISMISSED** [review]

File: app/views/layouts/_navbar.html.erb [hunk 5]
```diff
@@ -0,0 +1,39 @@
+<head>
+  <script src="http://code.jquery.com/jquery-1.11.0.min.js"></script>
```
Comment (YilKanda) [inline app/views/layouts/_navbar.html.erb:2]: The JQuery reference is no longer required, so you can remove it.

File: app/views/layouts/_navbar.html.erb [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+            <a class="nav-link" aria-current="page" href="#" id="linkContacto">Contacto</a>
+          </li>
+          
+        </div>
+          <div class="group609">
+            <%= image_tag "Group 609.svg", alt: "group609" %> 
+          </div>
+      </div>
+  </nav>
+</body>
```
Comment (YilKanda) [inline app/views/layouts/_navbar.html.erb:39]: Head and body tags are not necessary in this file, as they already exist in `application.erb.html`

```suggestion
  <nav class="navbar navbar-expand-lg navbar-light ">
    <div class="container-fluid">
      <div class="logo">
        <%= image_tag "BCnameLogo.png", alt: "BCLogo", class: "logo-closed", id:"logo" %> 
      </div>
      <button id="button" class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarTogglerDemo02"
        aria-controls="navbarTogglerDemo02" aria-expanded="false" aria-label="Toggle navigation" style="border: none; box-shadow: none">
        <%= image_tag "list.svg", alt: "BCLogo", id:"image" %> 
      </button>
      <div class="collapse navbar-collapse " id="navbarTogglerDemo02">
        <div class="navbar-nav ms-auto mb-2 mb-lg-0">
          <li class="nav-item">
            <a class="nav-link" aria-current="page" href="#" id="link">Nosotros</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" aria-current="page" href="#" id="link">Programas</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" aria-current="page" href="#" id="link">Graduados</a>
          </li>
          <li class="nav-item">
            <a class="nav-link" aria-current="page" href="#" id="link">Convocatorias</a>
          </li>
          <li class="nav-item" id="borderContacto">
            <a class="nav-link" aria-current="page" href="#" id="linkContacto">Contacto</a>
          </li>
          
        </div>
          <div class="group609">
            <%= image_tag "Group 609.svg", alt: "group609" %> 
          </div>
      </div>
  </nav>

  <%= javascript_pack_tag 'navbar', 'data-turbolinks-track': 'reload' %>

```

**Review by YilKanda — CHANGES_REQUESTED** [review]

File: app/views/layouts/_navbar.html.erb (hunk 6 above)
Comment (AlejandroVI99) [reply app/views/layouts/_navbar.html.erb:39]: Ok! understood!

File: app/views/layouts/_navbar.html.erb (hunk 5 above)
Comment (AlejandroVI99) [reply app/views/layouts/_navbar.html.erb:2]: Ok! understood!

File: app/views/layouts/_navbar.html.erb (hunk 6 above)
Comment (SampabloC) [reply app/views/layouts/_navbar.html.erb]: @YilKanda ready! 👍

**Comment by cfcortes** [comment]
@SampabloC @AlejandroVI99  I found some details when reviewing running app on [heroku](https://bc-landing-2-feature-3--bhufnt.herokuapp.com/):

- Menu effect when mouse over is different from design:
  - Options seems to move but they should not move
  - Underline effect is not an underline style
- On mobile view logos disappear
-  I will ask design team how should the button look like when mouse over 

Let me know if you want to discuss this on google meet or other

**Review by YilKanda — COMMENTED** [review]
It only remains to solve what happens to the logo when the menu is open, for example on an iPad in portrait mode, and it changes to landscape mode.

imagen

imagen

**Review by xFerchoVJ — APPROVED** [review]
LGTM!
It is advisable to put the properties of your styles in alphabetical order, it is a good practice

**Review by YilKanda — APPROVED** [review]
LGTM!

**Pull request merged**